org-formation/090-systems-manager/Scheduled-Script-Automation.yaml

```diff
@@ -0,0 +1,297 @@
+# *
+# * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# * SPDX-License-Identifier: MIT-0
+# *
+# * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+# * software and associated documentation files (the "Software"), to deal in the Software
+# * without restriction, including without limitation the rights to use, copy, modify,
+# * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+# * permit persons to whom the Software is furnished to do so.
+# *
+# * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+# * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+# * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+# *
+AWSTemplateFormatVersion: '2010-09-09'
+Description: AWS CloudFormation template to create a scheduled multi-account and multi-Region script execution.
+Parameters:
+  EventBridgeRuleSchedule:
+    Type: String
+    Description: >-
+      The cron or rate expression to use for the EventBridge rule. For example: cron(0 02 ? * TUE *). Important: The time zone used is UTC. For more information, see https://docs.aws.amazon.com/eventbridge/latest/userguide/scheduled-events.html.
+  ExecutionRoleName:
+    Type: String
+    Default: AWS-SystemsManager-AutomationExecutionRole
+    Description: >-
+      The Automation execution role to be assumed in target accounts during multi-account and multi-Region Automation operations.
+  ExistingAutomationAdministrationRole:
+    Type: String
+    Default: ''
+    Description: >-
+      (Optional) The IAM role ARN of an existing Automation Administration role which has permissions to invoke multi-account and multi-Region Automation workflows.
+  MaximumConcurrency:
+    Type: String
+    Default: 10%
+    Description: >-
+      Specify the number or percentage of targets on which to execute the task at the same time. You can specify a number, such as 10, or a percentage, such as 10%. The default value is 10%.
+  MaximumErrors:
+    Type: String
+    Default: 10%
+    Description: >-
+      The number of errors that are allowed before the system stops initiating the automation on additional targets. You can specify either an absolute number of errors, for example 10, or a percentage of the target set, for example 10%. The default value is 10%.
+  ScriptUrl:
+    Type: String
+    Description: The URL of the script to run on targeted instances. Must be publicly readable.
+  ScriptParameters:
+    Type: String
+    Description: >-
+      space delimited list of environment name-value pairs in the format name:value,
+      e.g., ENVNAME1:envval1 ENVNAME2:envval2
+  TargetTagName:
```

Comment:
I'm not sure why you are asking this question, since your own solution for patching relies on [tags](https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/org-formation/090-systems-manager/patch-resource-group.yaml#L23-L29) as well.  Moreover, all options I've seen for targeting resources with SSM automation either depend on tags or on all the resources being in the same CF stack.  But to answer your question:  My thought is to have the Synapse team add the required tag to the instances deployed by the Synapse stack builder and if we want Stack Armor to scan any machines besides the Synapse EC2s then we would include the tag in the deployment infra' for those stacks as well.

Reply:
> depending on human users to apply the right tag every time they deploy a resource is brittle at best

The expectation is not that human users apply the tag manually but that deployment (CF) scripts do it automatically.  I didn't realize that was not clear, so I have now tried to clarify that in the comments.

Reply:
> if someone deploys with a template that does not set the correct tags?

Then the automation will not run on the instance.  

I feel like you are trying to hint at an alternative.  If so, will you say what it is?

Reply:
my suggestion was in my previous comment, 
```In this use case I think a better solution is to make AWS automatically tag every EC2 instance with a special tag (i.e. nessus-monitoring) then have this solution install the nessus agent on those instances. This would mitigate the brittle part of this intended solution.```

to `make AWS automatically tag` resources you can use the [AWS tag editor](https://docs.aws.amazon.com/tag-editor/latest/userguide/tagging.html), a script using the AWS CLI, or a script using the SDK.  Those automation options can be run with a lambda or some type of cron job.

Reply:
>  make AWS automatically tag every EC2 instance with a special tag (i.e. nessus-monitoring) then have this solution install the nessus agent on those instances.

Yes, that is exactly my intention:  The [Synapse Stack Builder](https://github.com/Sage-Bionetworks/Synapse-Stack-Builder) will be changed to add the tag to all instances that we wish to monitor.  Since the change will be in that repo' you will not see it in this PR.